### DISCUSSION TITLE
[Feature Request] Display purchase button for claimed Pro games

### DISCUSSION THREAD

**Discussion by M-YD** [ChristopherKlay/StadiaEnhanced#182] [Ideas & Requests]
Hello,

It would be great to have the option to purchase claimed Pro games.

I am a Founder, yet have only become a **paying** Pro subscriber for the first time in October 2021 because the Pro games on offer up until now weren't tempting enough to make me bite.

The reason I would like to be able to purchase these games outright (especially when on sale, for example), is because I’m fairly sure that I won't keep my paid subscription active indefinitely, yet whilst I am an active Pro subscriber, I do not have the option to purchase a game and will therefore miss out on potential savings whilst active.

As such, the option to have both the "Claim" button alongside the "Buy for £xxx" would be ideal because, even though I have claimed a game on Pro, it would be great to be able to also buy the game should I choose to do so within my active subscription.

For clarity, already-claimed games’ pages would display a “Play” button alongside a “Buy for £xxx” button.

I’m happy to explain further if necessary.

**Comment by ChristopherKlay** [comment]
While i would love to provide a solution for this, it's not as easy as it sounds.

The main problem here is that any "purchase" has a unique ID (including sales, Pro sales and such), that isn't present on the website, unless the purchase button element is. Due to not having access to the purchase ID, i can't just create a button to use the given ID to start the purchase dialog.

The only (hacky) solution is the recently updated guide available in the wiki of this GitHub [here](https://github.com/ChristopherKlay/StadiaEnhanced/wiki/Buy-claimed-games) that allows you to copy the purchase options (including their ID's) from a different account. It's still limited (Pro deals still need Pro on the given account to be copied, for example), but it's currently the best solution i'm aware off.

**Reply by M-YD** [reply]
@ChristopherKlay I understand and appreciate the limitations here.

This suggestion came about following a [YouTube video by Gem](https://www.youtube.com/watch?v=Z3thyZb-bbA) where he describes how to achieve this and is very similar to the article you linked to; it would be great to have this implemented in some capacity as I'm sure that a fair percentage of Stadia users would benefit from it.

Given these limitations, what are your thoughts?  Is it something you think could be incorporated, perhaps in beta?

**Reply by ChristopherKlay** [reply]
Implementing it would basically mean keeping a always up-to-date database of every possible purchase option (normal, base sale, pro sale, ..), for every single game, some form of validation if the given sale even exists at the current time and a way to avoid Stadia disabling the "faked" purchase option after a while, after it's copied in (which is why you need to refresh the page, before copying the code in, in the how-to i linked).

Based on that, there currently isn't any feasible way to implement this. Any "solution" could also push the extension into grey area when it comes to Googles allowing it (due to being purchase related), which is something i'd like to avoid.
